Core/Src/bmbUtils.c

```diff
@@ -129,6 +147,35 @@ float interpolate(float x, float x1, float x2, float y1, float y2)
     }
 }
 
+/*!
+  @brief   Search a sorted array of Brick_S structs using binary search
+  @param   arr - Pointer to the Brick_S array to be searched
+  @param   l - Index of the left element of the array 
+  @param   r - Index of the right element of the array
+  @param   v - The target voltage to be compared to the Brick_S struct voltage
+  @return  Index at which voltage is to be inserted
+*/
+int brickBinarySearch(Brick_S *arr, int l, int r, float v)
+{
+  while (l <= r)
+  {
+    int m = l + (r - l) / 2;
```

Comment:
Not sure how often you call this but a >>1 is much faster than a /2

Reply:
True but I'm pretty sure that even with `-O0` the compiler will do that optimization for me

Reply:
Probably. I actually would be curious to know if you want to check the disassembly

Reply:
https://godbolt.org/z/f7YcT3hzc
With our optimization `-Og` level it does. No divide instruction to be found with `-O0'